docs/topics/control-flow.md

```diff
@@ -50,71 +50,152 @@ val max = if (a > b) {
 If you're using `if` as an expression, for example, for returning its value or
 assigning it to a variable, the `else` branch is mandatory.
 
-## When expression
+## When expressions and statements
 
-`when` defines a conditional expression with multiple branches. It is similar to the `switch` statement in C-like languages.
-Its simple form looks like this.
+`when` is a conditional expression that executes code based on multiple possible values or conditions. It is
+similar to the `switch` statement in Java, C, and similar languages. For example:
 
 ```kotlin
+fun main() {
+    //sampleStart
+    val x = 2
+    when (x) {
+        1 -> print("x == 1")
+        2 -> print("x == 2")
+        else -> print("x is neither 1 nor 2")
+    }
+    //sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-conditions-when-statement"}
+
+`when` matches its argument against all branches sequentially until some branch condition is satisfied.
+
+You can use `when` in a few different ways. Firstly, you can use `when` either as an **expression** or as a **statement**.
+As an expression, `when` returns a value that you can use later in your code. As a statement, `when` completes an action
+without returning anything of further use:
+
+<table>
+   <tr>
+       <td>Expression</td>
+       <td>Statement</td>
+   </tr>
+   <tr>
+<td>
+
+```kotlin
+// Returns a string assigned to the 
+// text variable
+val text = when (x) {
+    1 -> "x == 1"
+    2 -> "x == 2"
+    else -> "x is neither 1 nor 2"
+}
+```
+
+</td>
+<td>
+
+```kotlin
+// Returns nothing but triggers a 
+// print statement
 when (x) {
     1 -> print("x == 1")
     2 -> print("x == 2")
-    else -> {
-        print("x is neither 1 nor 2")
-    }
+    else -> print("x is neither 1 nor 2")
 }
 ```
 
-`when` matches its argument against all branches sequentially until some branch condition is satisfied.
+</td>
+</tr>
+</table>
 
-`when` can be used either as an expression or as a statement. If it is used as an expression, the value
-of the first matching branch becomes the value of the overall expression. If it is used as a statement, the values of
-individual branches are ignored. Just like with `if`, each branch can be a block, and its value
-is the value of the last expression in the block.
+Secondly, you can use `when` with or without a subject.
```

Comment:
There's the note below which explains that we recommend using a subject to make it easier to read and maintain your code. Even further below, we explain that a subject with an expression has the advantage that you don't have to use an `else` branch. Basically, using a subject with expressions makes the compiler's life easier because it just has to check the subject. I didn't add this detail because explaining _why_ it's easier leads to describing the inner mechanics of the compiler 😅

Reply:
I see. So the behavior is the same, and the same goal can be achieved with and without subject, but adding a subject is a choice that makes the life of the compiler easier, right? 
I think the gap we can bridge is clarifying that having or not having a subject doesn't change/affect the output of the code and then reinforcing what is the best practice and why. What do you think? Otherwise, even that we have the note about the recommendation, when looking at the 2 options in the table, there's an open question about: which one should I use according to what my code must achieve (beyond the recommended practice) ?